src/snowflake/snowpark/dataframe.py

```diff
@@ -659,6 +801,7 @@ def stat(self) -> DataFrameStatFunctions:
     def analytics(self) -> DataFrameAnalyticsFunctions:
         return self._analytics
 
+    @dataframe_exception_handler
```

Comment:
From code maintainability perspective, do you think it would make sense to integrate the lineage functionality into the existing `@publicapi`, or would it be better to keep them separate?

Reply:
I thought about this but it is only applicate to dataframe API and not all public apis. @sfc-gh-lspiegelberg do you think it makes sense to add this in @publicapi?

Reply:
we could also do something like this

```
from snowflake.snowpark._internal.utils import publicapi as common_publicapi


def publicapi(func):
    @dataframe_exception_handler
    @common_publicapi
    def combined_wrapper(*args, **kwargs):
        return func(*args, **kwargs)
    return combined_wrapper
```

This will reduce the amount of overall change but it also increases the congitive load on being aware that the publicapi in dataframe has an attached `dataframe_exception_handler` attached. I do not think is the best approach here.

Reply:
We already handle some errors here https://github.com/snowflakedb/snowpark-python/blob/dc8ac301bc66568041b08a04505bd200629c7fcf/src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py#L148. Is it possible to combine?

Reply:
this is a better place. It would also capture other classes as well.